addon/-private/data-view/utils/scroll-handler.js

```diff
@@ -36,27 +36,23 @@ export class ScrollHandler {
         handlers
       };
       // TODO add explicit test
-      if (SUPPORTS_PASSIVE) {
+      if (this.isUsingPassive) {
         cache.passiveHandler = function() {
           ScrollHandler.triggerElementHandlers(element, cache);
         };
+
+        element.addEventListener('scroll', cache.passiveHandler, { capture: true, passive: true });
+      // TODO add explicit test
       } else {
         cache.passiveHandler = UNDEFINED_VALUE;
+
+        this.poll();
       }
     } else {
       cache = this.handlers[index];
       handlers = cache.handlers;
       handlers.push(handler);
     }
-
-    // TODO add explicit test
-    if (this.isUsingPassive && handlers.length === 1) {
```

Comment:
this block was moved to a similar `if/else` above, just in order to cleanup a bit.

However, since I did it some time ago, I don't remember exactly why I've ignored the `handlers.length === 1` part. Will need to take another look at it.

Reply:
So the source of error was exactly the `handlers.length === 1` check.

When we add the second handler to the same scrollable target, we fallen back to the polling mode, which in its turns makes test waiters to freeze.